javascript-review.js

```diff
@@ -50,24 +67,49 @@
 
 // What is NPM? In your brief explanation refer to package.json.
 
+const npm = "A npm is a dependency which can be installed using 'npm install'. In whichever project an npm is called, it will create a package.json file which stores all the details for every subsequent npm called to the file."
+
 // 13.
 
 // What are Event Listeners? Gives some examples in your response.
 
+const eventListener = "An Event Listener is a tool that will activate a javascript function after a certain 'action' has been recognised, such as the push of a button, or the release of said button. It is used in front end work to call some function based upon the users actions."
+
 // 14.
 
 // Define a function called numMult that takes two number arguments and a callback. The function will multiply the two numbers and pass the result to the callback as an argument. Now call numMult and in the callback simply console.log out the argument that has been passed through.
 
+const numMult = (num1, num2, callback) => {
+    const answer = num1 * num2;
+    callback(answer)
+}
+
+numMult(2,4,print)
+
 // 15.
 
 // a.
-// Define a function called addNum with that takes two number arguments. In this function simply return the addition of these numbers. 
+// Define a function called addNum with that takes two number arguments. In this function simply return the addition of these numbers.
+
+const addNum = (num1, num2) => {
+    return num1 + num2
+}
 
 // b.
-// Write a second function called numsPlusFunct that takes three arguments, two numbers and a function. Inside numsPlusFunct call the function that is passed as an argument, and pass the two number arguments to this function. numsPlusFunct will return an object where the first property has the value returned from that function call, and the second property is a string. 
+// Write a second function called numsPlusFunct that takes three arguments, two numbers and a function. Inside numsPlusFunct call the function that is passed as an argument, and pass the two number arguments to this function. numsPlusFunct will return an object where the first property has the value returned from that function call, and the second property is a string.
+
+const numsPlusFunct = (num1, num2, callback) => {
+    const object = {
+        value: callback(num1,num2),
+        string: String(callback(num1,num2))
+    }
+    return object
+}
 
 // c.
-// You have now made two functions. Call the numsPlusFunct and pass addNum as the appropriate argument. 
+// You have now made two functions. Call the numsPlusFunct and pass addNum as the appropriate argument.
+
+print(numsPlusFunct(3,4,addNum))
```

Comment:
this is a really great answer, next time round for a review just be careful with re-using the same `print` function from above as it makes it hard to read your code